central/signatureintegration/datastore/datastore_impl.go

```diff
@@ -149,6 +158,45 @@ func (d *datastoreImpl) verifyIntegrationIDDoesNotExist(ctx context.Context, id
 	return nil
 }
 
+func (d *datastoreImpl) verifyIntegrationIDIsNotInPolicy(ctx context.Context, id string) error {
+	// We want to avoid deleting a signature integration which is referenced by any policy. If that is the case,
+	// stop the deletion of the policy and return an appropriate error message to the user.
+	policies, err := d.policyStore.GetAllPolicies(ctx)
+	if err != nil {
+		return errox.InvariantViolation.CausedBy(err.Error())
+	}
+
+	var policiesContainingID []string
+	for _, p := range policies {
+		if checkIfPolicyContainsID(id, p) {
+			policiesContainingID = append(policiesContainingID, p.GetName())
+		}
+	}
+
+	if len(policiesContainingID) != 0 {
+		return errox.ReferencedByAnotherObject.Newf("cannot delete signature integration %q since the "+
+			"following policies are referencing it: [%s]", id, strings.Join(policiesContainingID, ","))
```

Comment:
Keep in mind that you might not be able to show those policies or risk leaking something a user might not see. I think ideally you would:
* list those you can (scoped access, not applicable here currently but make sense to code defensively) if you can (permission)
* show there is/are some without disclosing the names if you can't